### ISSUE TITLE
"Thread terminated due to fatal error: [attrib buffer] Working buffer not big enough" being "fixed" with Accurate RSX Reservations Access

### ISSUE THREAD

**Issue by Augusto7743** [RPCS3/rpcs3#10861]
Since RPCS3 versions from 2018 I known few games crashing with the message

F SIG: Thread terminated due to fatal error: [attrib buffer] Working buffer not big enough, buffer_length=67108864 allocated=4185088 requested=3101384656 guard=4194304 largest_pool=62208
(in file ../rpcs3/Emu/RSX/VK/vkutils/../../Common/ring_buffer_helper.h:92, in function alloc<256>) (errno=11)

the values  buffer_length, allocated ,requested , guard and largest_pool are random bytes.

the crash is because not enough system memory available ?
Using Lubuntu 21.04 and RPCS3 running a game the total system memory allocated was 2.7 GB and the system has 4 GB RAM plus 2 GB swap.

If enabling Accurate RSX Reservations Access , driver wake up delay for a little value ( 20 ) and using few preferred spu threads in 10 times starting the same game 1 or 2 times crash with the message above.
Enabling Accurate RSX Reservations Access avoid crash.

ULTIMATE MARVEL VS. CAPCOM 3 [BLUS30787] and Teenage Mutant Ninja Turtles Mutants in Manhattan [NPEB02246] crash with the message above.
the logs are 1 Accurate RSX Reservations Access disabled and 2 Accurate RSX Reservations Access enabled with preferred spu treads value 3 and  driver wake up delay 20 game played from start to ending.

Another details is that crash in moments that the game not does any loading and not are games using very much PS3 resources.
see the screeshot. some times crash when the screen has few details how if has some wrongs codes running in emulation trying allocate memory to a task that no was need to do.

[RPCS31.zip](https://github.com/RPCS3/rpcs3/files/7148904/RPCS31.zip)

[RPCS32.zip](https://github.com/RPCS3/rpcs3/files/7148906/RPCS32.zip)

![screen](https://user-images.githubusercontent.com/21184987/132969972-34314f66-a451-4677-ae0c-c3370cabf79f.jpg)

**Comment by artofsound** [comment]
Yeah, Gran Turismo 6 crashes me after one of the first cup races as well. I can play the race, but at the reward screen, after receiving the credits for the game, it will crash.

`·F 0:07:33.946065 {RSX [0x19a7e40]} SIG: Thread terminated due to fatal error: [index buffer] Working buffer not big enough, buffer_length=16777216 allocated=857704 requested=4294967290 guard=65536 largest_pool=72344
(in file D:\a\1\s\rpcs3\Emu\RSX\Common\ring_buffer_helper.h:92[:24], in function alloc) (e=0x000001e7[487])`

[RPCS3_Gran_Turismo_6_working_buffer.zip](https://github.com/RPCS3/rpcs3/files/7185725/RPCS3_Gran_Turismo_6_working_buffer.zip)
[RPCS3_Gran_Turismo_6_working_buffer_2.zip](https://github.com/RPCS3/rpcs3/files/7185726/RPCS3_Gran_Turismo_6_working_buffer_2.zip)

With Accurate RSX Reservations Access I can't even start the race properly. It crashes mid race or even before starting it.
[RPCS3_RSX.zip](https://github.com/RPCS3/rpcs3/files/7185749/RPCS3_RSX.zip)
[RPCS3_RSX2.zip](https://github.com/RPCS3/rpcs3/files/7185750/RPCS3_RSX2.zip)

**Comment by Satan86** [comment]
GT6 crash can be worked around by enabling Disable FIFO Reordering in the debug tab of settings. It's already being tracked in #7913

**Comment by artofsound** [comment]
> 
> 
> GT6 crash can be worked around by enabling Disable FIFO Reordering in the debug tab of settings. It's already being tracked in #7913

You're right, thank you. Thought I had it already checked.

**Comment by Mrlinkwii** [comment]
@Augusto7743 i think part of this issue is that you have/had very weak hardware

**Comment by elad335** [comment]
Probably outdated & fixed, closing. Open a new issue if persists.
